api/mock_server/submission.py

```diff
@@ -11,8 +11,37 @@ def __eq__(self, other):
             and self.submission_json == other.submission_json
 
 
+class SubmissionProgress:
+    """A class for representing the progress of a submission"""
+    def __init__(self, progress: list):
+        self.data, status = [], progress[0].copy()
+        for update in progress:
+            status.update(update)
+            self.data.append(status.copy())
+
+
 class SubmissionScenario:
     """A class for representing a single possible submission scenario"""
-    def __init__(self, new_submission, submission_infos):
-        self.new_submission = new_submission
-        self.submission_infos = submission_infos
+    def __init__(self, submission_info, submission_progress):
+        self.submission_info = submission_info
+        self.submission_progress = submission_progress
+
+
+class TestProgress:
```

Comment:
Does this (and others in this file) need to be a class since it's always called as `TestProgress(...).data`? I think a function returning a dict would be simpler.

Didn't find a simple way to define custom JSON serializers either..